service/src/main/kotlin/fi/espoo/evaka/invoicing/service/InvoiceGenerator.kt

```diff
@@ -330,33 +334,51 @@ fun Database.Read.getAbsenceStubs(spanningRange: DateRange, categories: Collecti
         .toList()
 }
 
-data class PlacementStub(val unit: DaycareId, val type: PlacementType)
-
-data class Placements(
-    val period: DateRange,
-    val headOfFamily: PersonId,
-    val placements: List<Pair<ChildWithDateOfBirth, PlacementStub>>
+data class PlacementStub(
+    @Nested("child")
+    val child: ChildWithDateOfBirth,
+    val unit: DaycareId,
+    val type: PlacementType
 )
 
-internal fun Database.Read.getInvoiceablePlacements(
+private fun Database.Read.getInvoiceablePlacements(
     spanningPeriod: DateRange
-): List<Placements> {
-    data class PlacementRow(val dateRange: DateRange, val unitId: DaycareId, val type: PlacementType)
+): Map<ChildId, List<Pair<DateRange, PlacementStub>>> {
+    return createQuery(
+        // language=sql
+        """
+SELECT p.child_id, c.date_of_birth AS child_date_of_birth, u.id AS unit, daterange(p.start_date, p.end_date, '[]') AS date_range, p.unit_id, p.type
+FROM placement p
+JOIN person c ON p.child_id = c.id
+JOIN daycare u ON p.unit_id = u.id AND u.invoiced_by_municipality
+WHERE daterange(start_date, end_date, '[]') && :period
+AND p.type = ANY(:invoicedTypes::placement_type[])
+"""
+    )
+        .bind("period", spanningPeriod)
+        .bind("invoicedTypes", PlacementType.invoiced())
+        .map { rv -> rv.mapColumn<DateRange>("date_range") to rv.mapRow<PlacementStub>() }
+        .groupBy { it.second.child.id }
+}
 
+private fun Database.Read.getInvoiceableTemporaryPlacements(
+    spanningPeriod: DateRange
+): Map<PersonId, List<Pair<DateRange, PlacementStub>>> {
     val placements = createQuery(
         // language=sql
         """
-            SELECT p.child_id, daterange(p.start_date, p.end_date, '[]') AS date_range, p.unit_id, p.type FROM placement p
-            JOIN daycare u ON p.unit_id = u.id AND u.invoiced_by_municipality
-            WHERE daterange(start_date, end_date, '[]') && :period
-            AND p.type = ANY(:invoicedTypes::placement_type[])
-        """.trimIndent()
+SELECT p.child_id, c.date_of_birth AS child_date_of_birth, u.id AS unit, daterange(p.start_date, p.end_date, '[]') AS date_range, p.unit_id, p.type
+FROM placement p
+JOIN person c ON p.child_id = c.id
+JOIN daycare u ON p.unit_id = u.id AND u.invoiced_by_municipality
+WHERE daterange(start_date, end_date, '[]') && :period
+AND p.type = ANY(:types::placement_type[])
+"""
     )
         .bind("period", spanningPeriod)
-        .bind("invoicedTypes", PlacementType.invoiced())
-        .map { row -> row.mapColumn<ChildId>("child_id") to row.mapRow<PlacementRow>() }
-        .groupBy { (childId) -> childId }
-        .mapValues { it.value.map { (_, placements) -> placements } }
+        .bind("types", arrayOf(PlacementType.TEMPORARY_DAYCARE, PlacementType.TEMPORARY_DAYCARE_PART_DAY))
+        .map { rv -> rv.mapColumn<DateRange>("date_range") to rv.mapRow<PlacementStub>() }
+        .groupBy { it.second.child.id }
```

Comment:
It seems to me that the only difference between these two SQL queries is the value passed to `invoicedTypes`/`types`, so they should be merged I think.